Assets/Scripts/Utilities/SoundManager.cs

```diff
@@ -2,254 +2,127 @@
-	
-	[HideInInspector] public GameObject audioObjHolder;
-	[HideInInspector] public List<AudioSource> audioObjs;
+	AudioSource defaultSource;
 
-	public Dictionary<string, AudioClip> clipList = new Dictionary<string, AudioClip>();
-	public bool muted = false;
+    void Awake()
+    {
+		defaultSource = gameObject.AddComponent<AudioSource>();
+		defaultSource.playOnAwake = false;
 
-	public void SetMuted(bool mute)
-	{
-		muted = mute;
-	}
+		SetupPool( gameObject );
+    }
 
-	public void SetupPool(GameObject soundManagerObj)
+	public void SetupPool( GameObject soundManagerObj )
 	{
 		// Create object to hold sounders
-		audioObjHolder = new GameObject(poolName);
+		audioObjHolder = new GameObject( "AudioPool" );
 		audioObjHolder.transform.parent = soundManagerObj.transform;
+		
+		loopObjHolder = new GameObject( string.Concat( "AudioPool", "_Loops" ) );
+		loopObjHolder.transform.parent = soundManagerObj.transform;
 	}
-
+	
 	public AudioSource CreateAudioObj()
 	{
-		GameObject audioSourceObj = new GameObject(poolName, typeof(AudioSource));
+		GameObject audioSourceObj = new GameObject( string.Concat( "AudioPool", "_", audioObjs.Count.ToString() ), typeof( AudioSource ) );
 		audioSourceObj.transform.parent = audioObjHolder.transform;
-
-		audioObjs.Add(audioSourceObj.audio);
+		
+		audioObjs.Add( audioSourceObj.audio );
 		audioSourceObj.audio.playOnAwake = false;
 		
 		return audioSourceObj.audio;
 	}
-}
-
-public class SoundManager : SingletonBehaviour<SoundManager>
-{
-	[SerializeField] AudioPool[] audioPools;
-	[SerializeField] bool destroyOnLoad;
 
-    // Use this for initialization
-    void Awake()
+	public AudioSource Play3DSoundAtPosition( AudioSource sourceData, Vector3 position )
     {
-		foreach(AudioPool audioPool in audioPools)
+		foreach ( AudioSource audioSource in audioObjs )
 		{
-			foreach(AudioClip audioClip in audioPool.audioClips)
+			Debug.Log( audioSource );
+			if ( !audioSource.isPlaying )
 			{
-				audioPool.clipList.Add(audioClip.name, audioClip);
+		        audioSource.transform.position = position;
+				return PlayAudioObj( sourceData, audioSource );
 			}
-
-			audioPool.SetupPool(gameObject);
 		}
-    }
 
-	public AudioSource PlaySoundAtPosition(string clip, Vector3 position)
-    {
-		foreach(AudioPool audioPool in audioPools)
-		{
-			if(audioPool.poolType == AudioType.Sound)
-			{
-				foreach(AudioSource audioSource in audioPool.audioObjs)
-				{
-					if(!audioSource.isPlaying)
-					{
-				        audioSource.transform.position = position;
-						return PlayAudioObj(audioSource, clip);
-					}
-				}
-
-				AudioSource newSource = audioPool.CreateAudioObj();
-				newSource.transform.position = position;
-				audioPool.audioObjs.Add(newSource);
-				return PlayAudioObj(newSource, clip);
-			}
-		}
+		AudioSource newSource = CreateAudioObj();
+		newSource.transform.position = position;
+		audioObjs.Add( newSource );
 
-		return null;
+		return PlayAudioObj( sourceData, newSource );
```

Comment:
I'd like this function to be refactored so that it only has one `return` in it. Instead of returning form the loop, use the loop to find the first available obj. If there is none, create one, then set its position:

```
AudioSource source;

foreach ()
{
    // set source if an unused audio source is found
}

if ( !source )
{
    // create new audio source object and set source to reference it.
}

// set values and return source
```

This keeps the logical flow of the function clean and cuts down on code duplication. It's not super important for this particular function, but it's a good practice to maintain.